### PR TITLE
Add documentation for the Browser Log Collection

### PR DESCRIPTION
**Pull request by NBParis**
### What does this PR do?
Add documentation for the Browser log collection `private beta`

### Motivation
Private beta

### Preview link
https://docs-staging.datadoghq.com/nils/browser-log/logs/log_collection/web_browser

### CODE REVIEW

File: content/en/logs/log_collection/javascript.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+- link: "logs/faq/log-collection-troubleshooting-guide"
+  tag: "FAQ"
+  text: "Log Collection Troubleshooting Guide"
+---
+
+
+
+Send log to Datadog from web browsers or other Javascript client thanks to the Datadog Client-side JavaScript logging library.
+
+With the `browser-agent-core` library you are able to send log directory to Datadog from JS clients and leverage the following features:
```
Comment (bcaudan) [inline content/en/logs/log_collection/javascript.md]: `browser-agent-core` is maybe not relevant anymore since we change the file name to `datadog-logs`

Comment (NBParis) [reply content/en/logs/log_collection/javascript.md]: Good point, forgot to change the name of the library when I updated the link. Will do

File: content/en/logs/log_collection/javascript.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+* Use the library as a logger. Everything is forwarded to Datadog as JSON documents.
+* Add `context` and extra attributes.
+* Forward every JavaScript errors.
+* Forward JavaScript's console logs.
+* Track real client IP address and useragent.
+* Automatic bulk posts.
+
+## Get the Public API Key
+
+**Public API Keys are in Private Beta**
```
Comment (apigirl) [inline content/en/logs/log_collection/javascript.md]: how will they get access to the beta? a request to the support team?

File: content/en/logs/log_collection/javascript.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+For security reasons, [API keys][2] cannot be used as they would be exposed in the Javascript code. To collect Logs from web browsers, a [Public API keys][3] must be used.
+To manage your Public API keys, go to your [Datadog API configuration page][4] in the `Public API Key` section as shown here:
+
+{{< img src="logs/log_collection/public_key.png" style="width:80%;" alt="Public API Keys" responsive="true" >}}
+
+Read our [Public API keys documentation][3] to learn more about the restrictions that applies.
+
+## Configure the Javascript logger
+
+The following parameters can be used to configure the library to send logs to Datadog:
```
Comment (apigirl) [inline content/en/logs/log_collection/javascript.md]: this is my newby showing, but we should state where it is that you update these parameters

Comment (NBParis) [reply content/en/logs/log_collection/javascript.md]: The examples should be right below.

File: content/en/logs/log_collection/javascript.md (hunk 2 above)
Comment (NBParis) [reply content/en/logs/log_collection/javascript.md]: Yes request to support team.

File: content/en/logs/log_collection/javascript.md [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  tag: "Documentation"
+  text: "Perform Log Analytics"
+- link: "logs/faq/log-collection-troubleshooting-guide"
+  tag: "FAQ"
+  text: "Log Collection Troubleshooting Guide"
+---
+
+
+
+Send logs to Datadog from web browsers or other Javascript client thanks to the Datadog Client-side JavaScript logging library.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
Send logs to Datadog from web browsers or other Javascript client thanks to the Datadog `datadog-logs` Client-side JavaScript logging library.
```

File: content/en/logs/log_collection/javascript.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+With the `datadog-logs` library you are able to send log directory to Datadog from JS clients and leverage the following features:
+
+* Use the library as a logger. Everything is forwarded to Datadog as JSON documents.
+* Add `context` and extra attributes.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
* Add `context` and extra attributes to each logs sent.
```

File: content/en/logs/log_collection/javascript.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+* Forward every JavaScript errors.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
* Wraps and forwards every JavaScript errors automatically.
```

File: content/en/logs/log_collection/javascript.md [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+- link: "logs/faq/log-collection-troubleshooting-guide"
+  tag: "FAQ"
+  text: "Log Collection Troubleshooting Guide"
+---
+
+
+
+Send logs to Datadog from web browsers or other Javascript client thanks to the Datadog Client-side JavaScript logging library.
+
+With the `datadog-logs` library you are able to send log directory to Datadog from JS clients and leverage the following features:
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
With the `datadog-logs` library you are able to send log directory to Datadog from JS clients and leverage its following features:
```

File: content/en/logs/log_collection/javascript.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+* Use the library as a logger. Everything is forwarded to Datadog as JSON documents.
+* Add `context` and extra attributes.
+* Forward every JavaScript errors.
+* Forward JavaScript's console logs.
+* Track real client IP address and useragent.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
* Record real client IP address and user agent.
```

File: content/en/logs/log_collection/javascript.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+* Automatic bulk posts.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
* Optimised network usage with automatic bulk posts.
```

File: content/en/logs/log_collection/javascript.md (hunk 2 above)
Comment (l0k0ms) [reply content/en/logs/log_collection/javascript.md]:
```suggestion
**Public API Keys are in Private Beta** Reach out to [Datadog support team](/help) to get it enabled for your Account.
```

File: content/en/logs/log_collection/javascript.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+## Get the Public API Key
+
+**Public API Keys are in Private Beta** 
+
+For security reasons, [API keys][2] cannot be used as they would be exposed in the Javascript code. To collect Logs from web browsers, a [Public API keys][3] must be used.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
For security reasons, [API keys][2] cannot be used to configure the `datadog-logs` library as they would be exposed client side in the Javascript code. To collect Logs from web browsers, a [Public API keys][3] must be used.
```

File: content/en/logs/log_collection/javascript.md [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+To manage your Public API keys, go to your [Datadog API configuration page][4] in the `Public API Key` section as shown here:
+
+{{< img src="logs/log_collection/public_key.png" style="width:80%;" alt="Public API Keys" responsive="true" >}}
+
+Read our [Public API keys documentation][3] to learn more about the restrictions that applies.
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
Read the [Public API keys documentation][3] to learn more about the restrictions that applies.
```

File: content/en/logs/log_collection/javascript.md [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+## Configure the Javascript logger
+
+The following parameters can be used to configure the library to send logs to Datadog:
+
+* Set `isCollectingError` to false to turn off the automatic JS and console error collection
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
* Set `isCollectingError` to `false` to turn off the automatic JS and console error collection.
```

File: content/en/logs/log_collection/javascript.md [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+* Use `addGlobalContext` to add JSON attribute to all the generated logs
+* Set `publicApiKey` to the value of the Public api_key (**only public api_key can be used in this library**)
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
* Set `publicApiKey` to the value of the Public API key (**only public api_key can be used in this library**)
```

File: content/en/logs/log_collection/javascript.md [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+    <script>
+      // Set your Public API key
+      Datadog.init({
+        publicApiKey: '<PUBLIC_API_KEY>',
+        isCollectingError: true,
+    });
+
+      // OPTIONAL
+      // add global metadata attributes
+      Datadog.addGlobalContext({'session_id': '1234'});
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
      Datadog.addGlobalContext({'<META_KEY>': '<META_VALUE>'});
```

File: content/en/logs/log_collection/javascript.md [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
      Datadog.addGlobalContext({'<META_KEY>': '<META_VALUE>'});
```

File: content/en/logs/log_collection/javascript.md [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+...
+</html>
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Send a custom log entry
+
+It is then possible to send custom log entries thanks to the `log` function:
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
Send a custom log entries directly to Datadog with the `log` function:
```

File: content/en/logs/log_collection/javascript.md [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+
+```
+Datadog.log(<message>,<json_parameter>,<severity>)
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]:
```suggestion
Datadog.log(<MESSAGE>,<JSON_ATTRIBUTES>,<SEVERITY>)
```

Comment (l0k0ms) [reply content/en/logs/log_collection/javascript.md]: @NBParis, just to confirm, JSON_ATTRIBUTES can be a dump of a JSON object that will be send alongside of the message?

File: content/en/logs/log_collection/javascript.md [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+
+The accepted severity value are `debug`, `info`, `warn` or `error`.
+
+**Example:**
+
+```
+...
+<script>
+...
+Datadog.log('Button clicked', { name: 'buttonName' });
```
Comment (l0k0ms) [inline content/en/logs/log_collection/javascript.md]: Shouldn't it be `{ 'name': 'buttonName' }` ?

Comment (NBParis) [reply content/en/logs/log_collection/javascript.md]: Apparently not. Will triple check.

File: content/en/account_management/faq/api-app-key-management.md [hunk 19]
```diff
@@ -14,9 +14,18 @@ API keys are unique to your organization. An API key is required by the Datadog
...
 
-## Add
+## Public API keys
 
-To add a Datadog API or application key, navigate to [Integration -> APIs][1], enter a name for your key, and click **Create API key** or **Create Application Key**.
+<div class="alert alert-warning">
+Public API keys are in private beta. <a href="https://docs.datadoghq.com/help/">Reach out to support</a> to turn on this feature for your account.
+</div>
+
+Public API keys are unique to your organization. A public API key is required by the [web browser log collector][3] to submit logs to Datadog.
```
Comment (irabinovitch) [inline content/en/account_management/faq/api-app-key-management.md]: Is this accurate? We don't support other types of logs with this key?

File: content/en/account_management/faq/api-app-key-management.md [hunk 20]
```diff
@@ -14,9 +14,18 @@ API keys are unique to your organization. An API key is required by the Datadog
...
+Those keys can only be used to send web browser logs to Datadog.
```
Comment (irabinovitch) [inline content/en/account_management/faq/api-app-key-management.md]: s/Those/These/ @NBParis

Comment (irabinovitch) [reply content/en/account_management/faq/api-app-key-management.md]: Is this accurate? We don't support other types of logs with this key?

File: content/en/account_management/faq/api-app-key-management.md [hunk 21]
```diff
@@ -26,7 +35,7 @@ To add a Datadog API or application key, navigate to [Integration -> APIs][1], e
 
 ## Remove
 
-To remove a Datadog API or application key, navigate to [Integration -> APIs][1] and select the **Revoke** button next to the key you want to remove:
+To remove a Datadog API or application or public API key, navigate to [Integration -> APIs][1] and select the **Revoke** button next to the key you want to remove:
```
Comment (irabinovitch) [inline content/en/account_management/faq/api-app-key-management.md]: "To remove a Datadog API or application or public API key" should be "Datadog API, application, or public API key"

File: content/en/logs/log_collection/javascript.md [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+  tag: "Documentation"
+  text: "Perform Log Analytics"
+- link: "logs/faq/log-collection-troubleshooting-guide"
+  tag: "FAQ"
+  text: "Log Collection Troubleshooting Guide"
+---
+
+
+
+Send logs to Datadog from web browsers or other Javascript client thanks to the Datadog's `datadog-logs` client-side JavaScript logging library.
```
Comment (irabinovitch) [inline content/en/logs/log_collection/javascript.md]: s/thanks/with/

File: content/en/logs/log_collection/javascript.md [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+* Record real client IP addresses and user agents.
+* Optimised network usage with automatic bulk posts.
+
+## Get a Public API Key
+
+<div class="alert alert-warning">
+Public API keys are in private beta. <a href="https://docs.datadoghq.com/help/">Reach out to support</a> to turn on this feature for your account.
+</div>
+
+For security reasons, [API keys][2] cannot be used to configure the `datadog-logs` library as they would be exposed client-side in the JavaScript code. To collect logs from web browsers, a [public API key][3] must be used.
```
Comment (irabinovitch) [inline content/en/logs/log_collection/javascript.md]: configure seems like the wrong word here. I assume you can't use API keys at all for any purpose, the limitation ins't specific to configuration @NBParis

**Pull request merged**